File: C++/384. Shuffle an Array.cpp
class Solution {
public:
    vector<int> arr;
    int n;
    Solution(vector<int>& nums) {
		arr = nums;
		n = arr.size();
	}
    
    vector<int> reset() {
        return arr;
    }
    
    vector<int> shuffle() {
        vector<int> shuffle= arr;	
		int l = n;
		for(int i = n-1; i>=0; i--) {
			int p = rand()%l;
			swap(shuffle[i], shuffle[p]);
			l--;
		}
		return shuffle;
    }
};

/**
 * Your Solution object will be instantiated and called as such:
 * Solution* obj = new Solution(nums);
 * vector<int> param_1 = obj->reset();
 * vector<int> param_2 = obj->shuffle();
 */

File: Java/384. Shuffle an Array.java
class Solution {
    int[] arr;
    int n;
    public Solution(int[] nums) {
        arr = nums;
		n = arr.length;
    }
    
    public int[] reset() {
        return arr;
    }
    
    public int[] shuffle() {
        int[] shuffle= arr.clone();	
		int l = n;
		for(int i = n-1; i>=0; i--) {
			int p = new Random().nextInt(i+1);
            int temp = shuffle[i];
			shuffle[i] = shuffle[p];
            shuffle[p] = temp;
			l--;
		}
		return shuffle;
    }
}

/**
 * Your Solution object will be instantiated and called as such:
 * Solution obj = new Solution(nums);
 * int[] param_1 = obj.reset();
 * int[] param_2 = obj.shuffle();
 */
